src/java.desktop/windows/native/libawt/windows/awt_Component.cpp

```diff
@@ -6356,22 +6356,51 @@ jboolean AwtComponent::_NativeHandlesWheelScrolling(void *param)
     return result;
 }
 
-void AwtComponent::_SetParent(void * param)
-{
+void AwtComponent::_SetParent(void * param) {
     if (AwtToolkit::IsMainThread()) {
-        JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
-        SetParentStruct *data = (SetParentStruct*) param;
+        JNIEnv *env = (JNIEnv *) JNU_GetEnv(jvm, JNI_VERSION_1_2);
+        SetParentStruct *data = static_cast<SetParentStruct*>(param);
         jobject self = data->component;
         jobject parent = data->parentComp;
 
-        AwtComponent *awtComponent = NULL;
-        AwtComponent *awtParent = NULL;
+        AwtComponent *awtComponent = nullptr;
```

Comment:
Looking at it (not tested) here through 6403 could be simplified as

      if (self == NULL || parent == NULL) {
            env->ExceptionClear();
            JNU_ThrowNullPointerException(env, "peer");
            env->DeleteGlobalRef(self);
            env->DeleteGlobalRef(parent);
            delete data;
            return;
        }

        AwtComponent *awtComponent = (AwtComponent *)JNI_GET_PDATA(self);
        if (awtComponent == NULL) {
            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
            env->DeleteGlobalRef(self);
            env->DeleteGlobalRef(parent);
            delete data;
            return;
        }

        AwtComponent *awtParent = (AwtComponent *)JNI_GET_PDATA(parent);
        if (awtParent == NULL) {
            THROW_NULL_PDATA_IF_NOT_DESTROYED(parent);
            env->DeleteGlobalRef(self);
            env->DeleteGlobalRef(parent);
            delete data;
            return;
        }


I think I prefer that over
        BOOL error = FALSE;
        AwtComponent *awtComponent = nullptr;
        AwtComponent *awtParent = nullptr;

        if (self == NULL || parent == NULL) {
            env->ExceptionClear();
            JNU_ThrowNullPointerException(env, "peer");
            error = TRUE;
        }
        
        if (!error) {
            awtComponent = (AwtComponent *)JNI_GET_PDATA(self);
             if (awtComponent == NULL) {
                 THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
                 error = TRUE;
             }
        }

        if (!error) {
            awtParent = (AwtComponent *)JNI_GET_PDATA(parent);
            if (awtParent == NULL) {
                THROW_NULL_PDATA_IF_NOT_DESTROYED(parent);
                error = TRUE;
        }

        if (error) {
            env->DeleteGlobalRef(self);
            env->DeleteGlobalRef(parent);
            delete data;
            return;
        }

Reply:
Sorry, I don't see a BOOL error anywhere?

Reply:
I see the advantage of collapsing self and parent into the same check, but it doesn't seem like getting rid of pData is of much benefit, the checks for null seem to remain the same either way

Reply:
> Sorry, I don't see a BOOL error anywhere?

I think Phil misplaced the code block marker -- the BOOL error line and below is supposed to be inside the second code block.

What Phil is doing is trying to suggest two different approaches to make this code nicer, the first with a bit more duplication, and the second with an `error` boolean flag. And he says that he prefers the former, so the second were propably just mentioned to show an alternative for discussion.

I think what you should do here is to combine the `self` and `parent` null checks as Phil suggests. And then replace all pData references to the final variable, here as well as everywhere pData is involved.

Reply:
Ah, I see. Alright, will do so